### ISSUE TITLE
Hilfe benötigt (VWF 117/4 + VRC 720 + VR 921

### ISSUE THREAD

**Issue by Geg0r** [john30/ebusd-configuration#304]
Hi all,

I just received my rev3.1 ebusd adapter and now I am trying since hours to connect my Vaillant VWF 117/4 to it.

`root@ebusd:/etc/ebusd/config# ebusctl info
version: ebusd 23.1.23.1
device: 192.168.1.107:9999, enhanced, firmware 1.1[8419].2[847f]
signal: acquired
symbol rate: 24
max symbol rate: 115
min arbitration micros: 1
max arbitration micros: 2
min symbol latency: 7
max symbol latency: 10
reconnects: 0
masters: 5
messages: 509
conditional: 0
poll: 0
update: 10
address 03: master #11
address 08: slave #11, scanned "MF=Vaillant;ID=HMU00;SW=0308;HW=0403", loaded "vaillant/08.hmu.csv"
address 10: master #2
address 15: slave #2, scanned "MF=Vaillant;ID=72000;SW=0122;HW=7703", loaded "vaillant/15.720.csv"
address 31: master #8, ebusd
address 36: slave #8, ebusd
address 71: master #9
address 76: slave #9, scanned "MF=Vaillant;ID=VWZ00;SW=0308;HW=0403"
address f1: master #10
address f6: slave #10, scanned "MF=Vaillant;ID=NETX2;SW=4027;HW=5703"

`

From my point of view this looks good but the results are really low... hardly any results are found...

`root@ebusd:/etc/ebusd/config# ebusctl find
720 AdaptHeatCurve = no data stored
720 BankHolidayEndPeriod = no data stored
720 BankHolidayStartPeriod = no data stored
720 ccTimer.Friday = no data stored
720 ccTimer.Monday = no data stored
720 ccTimer.Saturday = no data stored
720 ccTimer.Sunday = no data stored
720 ccTimer.Thursday = no data stored
720 ccTimer.Tuesday = no data stored
720 ccTimer.Wednesday = no data stored
720 ContinuosHeating = no data stored
720 currenterror = no data stored
720 CylinderChargeHyst = no data stored
720 CylinderChargeOffset = no data stored
720 Date = no data stored
720 DisplayedOutsideTemp = 0.125
720 errorhistory = no data stored
720 FrostOverRideTime = no data stored
720 GlobalSystemOff = no data stored
720 Hc1ActualFlowTempDesired = no data stored
720 Hc1AutoOffMode = no data stored
720 Hc1CircuitType = no data stored
720 Hc1ExcessTemp = no data stored
720 Hc1FlowTemp = 24.5
720 Hc1HeatCurve = no data stored
720 Hc1HeatCurveAdaption = no data stored
720 Hc1MaxFlowTempDesired = no data stored
720 Hc1MinFlowTempDesired = no data stored
720 Hc1MixerMovement = no data stored
720 Hc1PumpStatus = 0
720 Hc1RoomTempSwitchOn = no data stored
720 Hc1Status = no data stored
720 Hc1SummerTempLimit = no data stored
720 Hc2ActualFlowTempDesired = no data stored
720 Hc2AutoOffMode = no data stored
720 Hc2CircuitType = no data stored
720 Hc2ExcessTemp = no data stored
720 Hc2FlowTemp = no data stored
720 Hc2HeatCurve = no data stored
720 Hc2HeatCurveAdaption = no data stored
720 Hc2MaxFlowTempDesired = no data stored
720 Hc2MinFlowTempDesired = no data stored
720 Hc2MixerMovement = no data stored
720 Hc2PumpStatus = no data stored
720 Hc2RoomTempSwitchOn = no data stored
720 Hc2Status = no data stored
720 Hc2SummerTempLimit = no data stored
720 Hc3ActualFlowTempDesired = no data stored
720 Hc3AutoOffMode = no data stored
720 Hc3CircuitType = no data stored
720 Hc3ExcessTemp = no data stored
720 Hc3FlowTemp = no data stored
720 Hc3HeatCurve = no data stored
720 Hc3HeatCurveAdaption = no data stored
720 Hc3MaxFlowTempDesired = no data stored
720 Hc3MinFlowTempDesired = no data stored
720 Hc3MixerMovement = no data stored
720 Hc3PumpStatus = no data stored
720 Hc3RoomTempSwitchOn = no data stored
720 Hc3Status = no data stored
720 Hc3SummerTempLimit = no data stored
720 HcStorageTempBottom = no data stored
720 HcStorageTempTop = no data stored
720 HolidayEndPeriod = no data stored
720 HolidayStartPeriod = no data stored
720 HolidayTemp = no data stored
720 HwcBankHolidayEndPeriod = no data stored
720 HwcBankHolidayStartPeriod = no data stored
720 HwcFlowTemp = no data stored
720 HwcHolidayEndPeriod = no data stored
720 HwcHolidayStartPeriod = no data stored
720 HwcLockTime = no data stored
720 HwcMaxFlowTempDesired = no data stored
720 HwcOpMode = day
720 HwcParallelLoading = no data stored
720 HwcSFMode = no data stored
720 HwcStorageTemp = 50
720 HwcStorageTempBottom = no data stored
720 HwcStorageTempTop = no data stored
720 HwcTempDesired = 49
720 hwcTimer.Friday = no data stored
720 hwcTimer.Monday = no data stored
720 hwcTimer.Saturday = no data stored
720 hwcTimer.Sunday = no data stored
720 hwcTimer.Thursday = no data stored
720 hwcTimer.Tuesday = no data stored
720 hwcTimer.Wednesday = no data stored
720 HydraulicScheme = no data stored
720 Installer1 = no data stored
720 Installer2 = no data stored
720 KeyCodeforConfigMenu = no data stored
720 MaintenanceDate = no data stored
720 MaintenanceDue = no data stored
720 ManualCooling = no data stored
720 MaxCylinderChargeTime = no data stored
720 MaxRoomHumidity = no data stored
720 MultiRelaySetting = no data stored
720 noiseReductionTimer.Friday = no data stored
720 noiseReductionTimer.Monday = no data stored
720 noiseReductionTimer.Saturday = no data stored
720 noiseReductionTimer.Sunday = no data stored
720 noiseReductionTimer.Thursday = no data stored
720 noiseReductionTimer.Tuesday = no data stored
720 noiseReductionTimer.Wednesday = no data stored
720 OpMode = no data stored
720 OpModeCooling = no data stored
720 OpModeEffect = no data stored
720 OpModeVentilation = no data stored
720 OutsideTempAvg = no data stored
720 PhoneNumber1 = no data stored
720 PhoneNumber2 = no data stored
720 PrEnergySum = no data stored
720 PrEnergySumHc = no data stored
720 PrEnergySumHcLastMonth = no data stored
720 PrEnergySumHcThisMonth = no data stored
720 PrEnergySumHwc = no data stored
720 PrEnergySumHwcLastMonth = no data stored
720 PrEnergySumHwcThisMonth = no data stored
720 PrFuelSum = no data stored
720 PrFuelSumHc = no data stored
720 PrFuelSumHcLastMonth = no data stored
720 PrFuelSumHcThisMonth = no data stored
720 PrFuelSumHwc = no data stored
720 PrFuelSumHwcLastMonth = no data stored
720 PrFuelSumHwcThisMonth = no data stored
720 PumpAdditionalTime = no data stored
720 SFMode = no data stored
720 SolarYieldTotal = no data stored
720 SystemFlowTemp = -
720 tariffTimer.Friday = no data stored
720 tariffTimer.Monday = no data stored
720 tariffTimer.Saturday = no data stored
720 tariffTimer.Sunday = no data stored
720 tariffTimer.Thursday = no data stored
720 tariffTimer.Tuesday = no data stored
720 tariffTimer.Wednesday = no data stored
720 Time = no data stored
720 VentilationDay = no data stored
720 VentilationNight = no data stored
720 ventilationTimer.Friday = no data stored
720 ventilationTimer.Monday = no data stored
720 ventilationTimer.Saturday = no data stored
720 ventilationTimer.Sunday = no data stored
720 ventilationTimer.Thursday = no data stored
720 ventilationTimer.Tuesday = no data stored
720 ventilationTimer.Wednesday = no data stored
720 WaterPressure = 1.8
720 YieldTotal = no data stored
720 z1ActualRoomTempDesired = 0.0
720 z1BankHolidayEndPeriod = no data stored
720 z1BankHolidayStartPeriod = no data stored
720 z1CoolingTemp = no data stored
720 z1CoolingTimer.Friday = no data stored
720 z1CoolingTimer.Monday = no data stored
720 z1CoolingTimer.Saturday = no data stored
720 z1CoolingTimer.Sunday = no data stored
720 z1CoolingTimer.Thursday = no data stored
720 z1CoolingTimer.Tuesday = no data stored
720 z1CoolingTimer.Wednesday = no data stored
720 z1DayTemp = no data stored
720 z1HolidayEndPeriod = no data stored
720 z1HolidayStartPeriod = no data stored
720 z1HolidayTemp = no data stored
720 z1Name1 = no data stored
720 z1Name2 = no data stored
720 z1NightTemp = no data stored
720 z1OpMode = auto
720 z1OpModeCooling = no data stored
720 z1QuickVetoTemp = no data stored
720 z1RoomTemp = 19.2875
720 z1RoomZoneMapping = no data stored
720 z1SFMode = no data stored
720 z1Shortname = no data stored
720 z1Timer.Friday = no data stored
720 z1Timer.Monday = no data stored
720 z1Timer.Saturday = no data stored
720 z1Timer.Sunday = no data stored
720 z1Timer.Thursday = no data stored
720 z1Timer.Tuesday = no data stored
720 z1Timer.Wednesday = no data stored
720 z1ValveStatus = no data stored
720 z2ActualRoomTempDesired = no data stored
720 z2BankHolidayEndPeriod = no data stored
720 z2BankHolidayStartPeriod = no data stored
720 z2CoolingTemp = no data stored
720 z2CoolingTimer.Friday = no data stored
720 z2CoolingTimer.Monday = no data stored
720 z2CoolingTimer.Saturday = no data stored
720 z2CoolingTimer.Sunday = no data stored
720 z2CoolingTimer.Thursday = no data stored
720 z2CoolingTimer.Tuesday = no data stored
720 z2CoolingTimer.Wednesday = no data stored
720 z2DayTemp = no data stored
720 z2HolidayEndPeriod = no data stored
720 z2HolidayStartPeriod = no data stored
720 z2HolidayTemp = no data stored
720 z2Name1 = no data stored
720 z2Name2 = no data stored
720 z2NightTemp = no data stored
720 z2OpMode = no data stored
720 z2OpModeCooling = no data stored
720 z2QuickVetoTemp = no data stored
720 z2RoomTemp = no data stored
720 z2RoomZoneMapping = no data stored
720 z2SFMode = no data stored
720 z2Shortname = no data stored
720 z2Timer.Friday = no data stored
720 z2Timer.Monday = no data stored
720 z2Timer.Saturday = no data stored
720 z2Timer.Sunday = no data stored
720 z2Timer.Thursday = no data stored
720 z2Timer.Tuesday = no data stored
720 z2Timer.Wednesday = no data stored
720 z2ValveStatus = no data stored
720 z3ActualRoomTempDesired = no data stored
720 z3BankHolidayEndPeriod = no data stored
720 z3BankHolidayStartPeriod = no data stored
720 z3CoolingTemp = no data stored
720 z3DayTemp = no data stored
720 z3HolidayEndPeriod = no data stored
720 z3HolidayStartPeriod = no data stored
720 z3HolidayTemp = no data stored
720 z3Name1 = no data stored
720 z3Name2 = no data stored
720 z3NightTemp = no data stored
720 z3OpMode = no data stored
720 z3OpModeCooling = no data stored
720 z3QuickVetoTemp = no data stored
720 z3RoomTemp = no data stored
720 z3RoomZoneMapping = no data stored
720 z3SFMode = no data stored
720 z3Shortname = no data stored
720 z3ValveStatus = no data stored
broadcast datetime = no data stored
broadcast error = no data stored
broadcast hwcStatus = no data stored
broadcast id = no data stored
broadcast id = no data stored
broadcast load = no data stored
broadcast outsidetemp = 0.125
broadcast signoflife = no data stored
broadcast vdatetime = 00:48:58;27.01.2023
general valuerange = no data stored
hmu ConsumptionThisYear10 = no data stored
hmu ConsumptionThisYear11 = no data stored
hmu ConsumptionThisYear12 = no data stored
hmu ConsumptionThisYear1 = no data stored
hmu ConsumptionThisYear2 = no data stored
hmu ConsumptionThisYear3 = no data stored
hmu ConsumptionThisYear4 = no data stored
hmu ConsumptionThisYear5 = no data stored
hmu ConsumptionThisYear6 = no data stored
hmu ConsumptionThisYear7 = no data stored
hmu ConsumptionThisYear8 = no data stored
hmu ConsumptionThisYear9 = no data stored
hmu ConsumptionTotal = no data stored
hmu CurrentCompressorUtil = no data stored
hmu CurrentConsumedPower = no data stored
hmu currenterror = no data stored
hmu CurrentYieldPower = no data stored
hmu DateTime = valid;00:49:01;27.01.2023;0.125
hmu EnergyCool = no data stored
hmu EnergyHc = no data stored
hmu errorhistory = no data stored
hmu Hours = no data stored
hmu HoursCool = no data stored
hmu HoursHc = no data stored
hmu SetMode = auto;0.0;-;-;1;1;1;0;0;0
hmu State = no data stored
hmu Status01 = 24.5;23.5;0.125;-;50.0;off
hmu Status02 = no data stored
hmu Status16 = no data stored
hmu Status = no data stored
hmu StatusCirPump = on
hmu WaterThroughput = no data stored
hmu YieldThisYear10 = no data stored
hmu YieldThisYear11 = no data stored
hmu YieldThisYear12 = no data stored
hmu YieldThisYear1 = no data stored
hmu YieldThisYear2 = no data stored
hmu YieldThisYear3 = no data stored
hmu YieldThisYear4 = no data stored
hmu YieldThisYear5 = no data stored
hmu YieldThisYear6 = no data stored
hmu YieldThisYear7 = no data stored
hmu YieldThisYear8 = no data stored
hmu YieldThisYear9 = no data stored
hmu YieldTotal = no data stored
memory eeprom = no data stored
memory ram = no data stored
scan id = no data stored
scan.08  = Vaillant;HMU00;0308;0403
scan.08 id = 21;21;04;0010016422;0001;005082;N0
scan.15  = Vaillant;72000;0122;7703
scan.15 id = 21;21;03;0020260914;0953;006556;N3
scan.76  = Vaillant;VWZ00;0308;0403
scan.76 id =  (ERR: invalid position for 3176b5090124 / 00)
scan.f6  = Vaillant;NETX2;4027;5703
scan.f6 id = 21;21;06;0020260962;0938;009604;N2
`

I am mainly interessted in the hmu results like "hmu CurrentConsumedPower" but I am not able to receive this info.

What am I doing wrong?

Hopefully somebody can help.

Thanks and best regards
Geg0r

**Comment by cyberthom42** [comment]
you need to actively read the parameters. 
Did you try:  `ebusctl read CurrentConsumedPower` ?

**Comment by Geg0r** [comment]
Yes you are right this needs to be done and I change the 08.hmu.csv file replacing r with r1 to read the parameters every cycle. Data is now written to an influxes via mqtt but there are still a lot parameters missing

**Comment by cyberthom42** [comment]
there are improved csv files around. I actually put my own locally as some of the pull requests are still not incorporated. Yesterday I also figured out how to decode the parameter from the configuration menu "Fachhandwerkerebene" of the arotherm. But I gess it should be similar to yours. You can have a look at my github repository, I will also upload my files there

**Comment by Geg0r** [comment]
Yeah I also realized that there are different file flying around and I also created a local copy and putted all working parameters in my local file. But there are still a lot parameters missing. 
I still have 50 parameters which are shown in `ebusctl grab result` as unknown but I am struggeling with decoding them.
I will check your repo

**Comment by cyberthom42** [comment]
Yesterday I found out how to decode and change the settings at the "Fachhandwerkerebene", maybe this helps so far. I will upload this soon:
`# B5 1A Fachhandwerker Konfiguration,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
*r,,,,,,"B51A","05FF34",,,IGN:3,,,,,,,,,,,,,,,,,,,,,
*w,,,,,,"B51A","06FF34",,,,,,,,,,,,,,,,,,,,,,,,
r;w,,Restfoerderhoehe,,,,,1f,,,UIN,,,,,,,,,,,,,,,,,,,,,
r;w,,hwcBetriebsart,,,,,44,,,UIN,,,Betriebsart Warmwasser 0=eco 1=normal 2=balance,,,,,,,,,,,,,,,,,,
`

**Comment by cyberthom42** [comment]
How did you figure out that r1 means that to read the parameters every cycle. What is a cycle exactly (each read request?). I could not found a documentation about this.

**Comment by Geg0r** [comment]
The r1 thing was mentioned by @Thoro here: 
![image](https://user-images.githubusercontent.com/95048756/216610866-1802c79b-e171-43a9-b4b7-10f77ccd08d4.png)

**Comment by Thoro** [comment]
@cyberthom42 Documentation is here: https://github.com/john30/ebusd/wiki/4.1.-Message-definition

**Comment by highshot1** [comment]
Got the same configuration like you Geg0r. (vwf 117/4, VRC 720 + VR921)
thanks for your input. changed r to r1 and now i am getting few data... searching especially for brine temperatures.... still a long way to go to get there..
